config/schema/group_resolver_negotiation.schema.yml

```diff
@@ -0,0 +1,19 @@
+og.group_resolver_negotiation.*:
+  type: config_entity
+  label: 'OG group resolver negotiation config'
+  mapping:
+    id:
+      type: string
+      label: 'ID'
+    label:
+      type: label
+      label: 'Label'
+    description:
+      type: string
+      label: 'Description'
+    status:
+      type: boolean
+      label: 'Status'
+    weight:
+      type: integer
+      label: 'Weight'
```

Comment:
Is the intention to save a separate config instance for each plugin? Do you foresee that we will need to save multiple instances of the same plugin? I can imagine for example the query argument plugin to have a configurable keyword, and you might want to save multiple instances of it, so you can derive the group if the URL has the query argument `?og_group_ref=groupid` as well as `?group=groupid`.

To me this sound overkill though. I think it would be simpler to have every plugin available once, and manage them solely with their on/off status and their weight. That means we can save the entire negotiation config in simple config. You would ignore the disabled plugins, and simply list the enabled ones, ordered by weight.

Example config file: `config/install/og.group_resolvers.yml`
```
# Most important plugin: check if we are on a canonical entity route, like node/{node}.
route_canonical_entity
# Second, check the query argument, this is configurable.
query_argument:
  keyword: group
# A custom plugin provided by a project. E.g. checking if a group exists in an entity reference.
my_custom_plugin:
  reference_field: "myentity_myfield"
# Fallback plugin, iterate over all route parameters. This will provide group context on routes like edit forms etc.
route_parameters
```

This would make it a lot easier to manage.